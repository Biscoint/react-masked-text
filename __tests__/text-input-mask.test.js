// Link.react.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import TextInputMask from '../src/text-input-mask'

test('TextInputMask renders uncontrolled component', () => {
  const component = renderer.create(
    <TextInputMask kind='cpf' defaultValue='123'/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('TextInputMask renders controlled component', () => {
  const component = renderer.create( <TextInputMask kind='cpf' value={''}/>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component.update( <TextInputMask kind='cpf' value={'123123'}/>);
  const updatedTree = component.toJSON();
  expect(updatedTree).toMatchSnapshot();
});

test('TextInputMask renders controlled component with initial value', () => {
  const value = '223';
  const onChange = (val) => value = val;
  const component = renderer.create(
    <TextInputMask kind='cpf' onChangeText={onChange} value={value}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});