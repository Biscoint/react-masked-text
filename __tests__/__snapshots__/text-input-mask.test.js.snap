// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInputMask renders controlled component 1`] = `
<input
  kind="cpf"
  onChange={[Function]}
  value=""
/>
`;

exports[`TextInputMask renders controlled component 2`] = `
<input
  kind="cpf"
  onChange={[Function]}
  value="123.123"
/>
`;

exports[`TextInputMask renders controlled component with initial value 1`] = `
<input
  kind="cpf"
  onChange={[Function]}
  value="223"
/>
`;

exports[`TextInputMask renders uncontrolled component 1`] = `
<input
  kind="cpf"
  onChange={[Function]}
  value="123"
/>
`;
